pkg/controllers/nodeclass/status/k8sversion.go

```diff
@@ -0,0 +1,102 @@
+/*
+Portions Copyright (c) Microsoft Corporation.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package status
+
+import (
+	"context"
+	"fmt"
+
+	"knative.dev/pkg/logging"
+
+	"github.com/Azure/karpenter-provider-azure/pkg/apis/v1alpha2"
+	azurecache "github.com/Azure/karpenter-provider-azure/pkg/cache"
+	"github.com/Azure/karpenter-provider-azure/pkg/providers/imagefamily"
+	"github.com/awslabs/operatorpkg/reasonable"
+	"github.com/blang/semver/v4"
+
+	controllerruntime "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/karpenter/pkg/utils/pretty"
+)
+
+type K8sVersionReconciler struct {
```

Comment:
Does this need to be exported?

Reply:
Its used in tests, which are black-box with a separate package, but could use the `Reconciler` interface instead. Though you have a question if that needs to be exported too :)

Reply:
Could also make that one test clear-box instead.

Reply:
Ginkgo started complaining about putting both clearbox, and blackbox tests in the same area while using `BeforeSuite`. I'm not opposed to breaking apart our testing in this way. However, I think the added complexity to just make this struct private is not worth it here, so leaving this for the moment as is.